File: server/src/Models/DMToolkitModels/Instances/AbilityScoreInstance.cs
using DMToolkit.Models.Definitions;

namespace DMToolkit.Models.Instances;

public class AbilityScoreInstance : IDefinitionInstance<AbilityScoreDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public int Score { get; set; } = 0;
    public bool IsProficient { get; set; } = false;
    public string CharacterId { get; set; } = string.Empty;
    public Character? Character { get; set; } = null;
    public ICollection<SkillInstance> SkillInstances { get; set; } = new List<SkillInstance>();
    public string DefinitionId { get; set; } = string.Empty;
    public AbilityScoreDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/abilityScoreInstance.ts
import type { AbilityScoreDefinition } from "../definitions/abilityScoreDefinition";
import type { SkillInstance, SkillInstanceDto } from "./skillInstance";

export interface AbilityScoreInstance {
    id: string;
    score: number;
    isProficient: boolean;
    skillInstances: SkillInstance[];
    definition: AbilityScoreDefinition;
}

export interface AbilityScoreInstanceDto {
    id: string;
    score: number;
    isProficient: boolean;
    skillInstances: SkillInstanceDto[];
    definitionId: string;
}